### DISCUSSION TITLE
Modify action in ListenerMiddleware

### DISCUSSION THREAD

**Discussion by medihack** [reduxjs/redux-toolkit#4019] [Q&A] [answered]
Is it possible to modify an action in a ListenerMiddleware (in my case, attach some `meta` property)?

I tried it with something like below, but the action received by the reducer does not have the attached property.

```ts
listenerMiddleware.startListening({
  matcher: isAnyOf(action1, action2, action3),
  effect: (action, listenerApi) => {
    action.meta = { foo: "bar" }
  },
})
```

Is a ListenerMiddleware not suitable for that and I have to use some more [general Redux middleware](https://redux.js.org/understanding/history-and-design/middleware#seven-examples)?

**Comment by EskiMojo14** [comment, marked as answer]
listeners will always run *after* the action has already reached the reducers, so it's not possible to modify the action in them.

**Reply by medihack** [reply]
Ok, good to know. I will use a "normal" middleware then. Thanks for helping out again!
